test/caption-stream.test.js

```diff
@@ -844,6 +844,61 @@ QUnit.test('special and extended character codes work regardless of field and da
   QUnit.deepEqual(captions[2].text, String.fromCharCode(0xbb), 'CC4 extended character correct');
 });
 
+QUnit.test('number of roll up rows takes precedence over base row command', function(assert) {
+  var captions = [];
+  var packets = [
+    // RCL
+    { type: 0, ccData: 0x1420 },
```

Comment:
Any reason you're adding the RCL/EDM packets? You should be able to start right at the RU2

Reply:
Ah, I wanted the test to be as complete as possible; but I can change it either way

Reply:
Complete as in, [causing a flush within `setRollUp`](https://github.com/videojs/mux.js/blob/3e185285822d248779cc661de012c0211d7a7cd3/lib/m2ts/caption-stream.js#L726-L734)? IMO that's not needed.

Reply:
Yes, and to mimic how an actual caption would be received by the stream

Reply:
Why do you think an actual caption would need to start with RCL/EDM packets? It could have been in roll-up mode since "forever", in which case you wouldn't see RCL/EDM, just RU2/PAC (with valid row), and then at some point you'd see the switch to a PAC with an "invalid" row.

Reply:
Ah, you're right. I forgot that we change the mode on RU2/RU3 as well